### DISCUSSION TITLE
Tuple deconstruction in LINQ query comprehensions

### DISCUSSION THREAD

**Discussion by dcuccia** [dotnet/csharplang#7035] [Language Ideas]
Runtime Issue [#23917](https://github.com/dotnet/runtime/issues/23917) (Proposal [#23917](https://github.com/dotnet/runtime/issues/20167)) added deconstructing named tuples in `foreach` enumeration:
```csharp
foreach (var (x, y) in Seq1.Zip(Seq2))
```
A lambda-based implementation for the `Enumerable.Zip` selector is effectively already tracked in #258:
```csharp
var result = Seq1.Zip(Seq2, var (x, y) => ... );
```
But, the LINQ query comprehension syntax does not exist:
```csharp
var result = from (x, y) in Seq1.Zip(Seq2)) ...
```
I'm sorry if this is a duplicate - I could not immediately find an existing discussion or issue.

Edited 3/7 - thanks to @svick for the tip on #258

**Comment by svick** [comment]
The lambda part is a duplicate of https://github.com/dotnet/csharplang/discussions/258.

**Reply by dcuccia** [reply]
Thank you! I've updated the post and the title.
